tools/TestDBGenerator/Program.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace TestDBGenerator;
+
+using System.IO.Compression;
+using DurableTask.SqlServer;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Smo;
+
+static class Program
+{
+    static async Task Main(string[] args)
+    {
+        SqlConnectionStringBuilder builder = new(GetConnectionString());
+        Server dbServer = new(new ServerConnection(new SqlConnection(builder.ToString())));
+
+        // There's an assumption that the database schema version matches the assembly version.
+        Version assemblyVersion = typeof(SqlOrchestrationService).Assembly.GetName().Version!;
+        Version schemaVersion = new(assemblyVersion.Major, assemblyVersion.Minor, assemblyVersion.Build);
+
```

Comment:
can you elaborate on why we have this assumption? Is this something we should be checking against in this test? Does anything automatically go wrong if this assumption is not met?

Reply:
This is the behavior of our DB upgrade logic, which you can find [here](https://github.com/microsoft/durabletask-mssql/blob/a0a6a5f9f75d8a1fa5501c107c154a021f2cd7d8/src/DurableTask.SqlServer/SqlDbManager.cs#L30-L121). If this assumption is broken, then the upgrade test should fail because the database schema won't be upgraded properly. That said, we can probably add an assert that validates this explicitly.

Reply:
Got it. That file actually helped give me a lot of context around how this is supposed to work.
I think the test sufficiently checks for correctness, so I won't ask for an explicit check of this assumption.